### ISSUE TITLE
Incorrect errobar for drawstyle='steps-post'

### ISSUE THREAD

**Issue by MLopez-Ibanez** [mwaskom/seaborn#3702]
Plotting an errorbar for several step functions draws the error bars ignoring the steps:

```python
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
data = pd.DataFrame({
    "iteration" : [ 1, 3, 8,  1, 3, 8,  1, 3, 8],
    "value"     : [10, 9, 8, 11, 6, 5, 20, 5, 4],
    "run"       : [1,  1, 1,  2, 2, 2,  3, 3, 3],
})
sns.lineplot(data=data, x='iteration', y='value', errorbar='sd',drawstyle='steps-post')
plt.show()
```
![image](https://github.com/mwaskom/seaborn/assets/2620021/2829050b-5666-4cf0-bee4-b218922ba8ec)

It should be something like this:

![image](https://github.com/mwaskom/seaborn/assets/2620021/10c70c70-5995-4713-9c76-40521569fc5f)

using R:

```R
library(ggplot2)
library(pammtools)
data <- read.table(header=TRUE, text='
   iteration  value  run
          1     10    1
          3      9    1
          8      8    1
          1     11    2
          3      6    2
          8      5    2
          1     20    3
          3      5    3
          8      4    3
')
ggplot(data, aes(x=iteration, y=value)) +
  geom_step(stat = "summary", fun = mean) +
  stat_summary(geom='stepribbon', 
    fun.data = mean_se, alpha = 0.5)  # added transparency
```

Step functions arise in many real-world scenarios. See a more complicated example here: https://adibender.github.io/pammtools/reference/geom_stepribbon.html#examples

**Comment by mwaskom** [comment]
I don't think this is a bug really, just a consequence of `lineplot` accepting all of the parameters for the underlying matplotlib `plot` function, which has a vast functionality. So this is a feature request for `lineplot`, but probably not one that would get added any time soon.

**Comment by MLopez-Ibanez** [comment]
> I don't think this is a bug really, just a consequence of `lineplot` accepting all of the parameters for the underlying matplotlib `plot` function, which has a vast functionality. So this is a feature request for `lineplot`, but probably not one that would get added any time soon.

I can edit the title to mention Feature Request if you wish. 

Step functions are so common (timeseries, ECDFs, survival analysis, inventory levels, etc) that they may warrant a dedicated function `stairplot()` function. Perhaps this may inspire someone not willing to use R to implement it.
